frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -142,141 +173,218 @@ function ProblemDisplay(props: ProblemDisplayParams) {
   };
 
   return (
-    <Content>
-      <MediumText>Name:</MediumText>
-      <TextInput
-        name="name"
-        value={newProblem.name}
-        onChange={handleChange}
-      />
+    <>
+      <MainContent>
+        <FlexBareContainer>
+          <SmallHeaderText>Problem</SmallHeaderText>
+          <SmallButtonLink
+            color={ThemeConfig.colors.gradients.gray}
+            onClick={() => onClick(newProblem)}
+            to="/problems/all"
+            style={{ marginLeft: 'auto' }}
```

Comment:
Ties into my comment before but thoughts on moving some of the styling to the styled components? I know there's like a ton of small styling changes here and there which could make it difficult tho - not sure the best approach to something like that - maybe like a decent amount of styled components within this file at the top, idk

Reply:
For this case specificially though it might also help to put the two buttons in a new div together, and then make that div flex-right or something like that, that way you might not have to have an edge case style like marginLeft

Reply:
Maybe like a liberal use of styled-components specific to the type (e.g. TestCaseHeader) rather than style (PaddingTopContainer) might also help with some of the styling edge cases - might cause some extra components at the top though so I guess it's a tradeoff

Reply:
Yes for sure, I agree with this - I converted all of the elements to styled components rather than inline CSS, let me know what you think 👍